### ISSUE TITLE
Implement the ability to duplicate mesh directly from the mesh table

### ISSUE THREAD

**Issue by Pear0533** [Pear0533/FLVER_Editor#114]

**Comment by MassKlaus** [comment]
Hey there! I was looking at the new duplicate feature and noticed that if an item is Hidden then duplicated it will make things go a little wacky the duplicated item either never being visible or being inverted. 

I just wanted to ensure some behavior before I could make a pull request with a fix (mostly one line)

after duplication, should all models be unselected and unhidden?

**Comment by Pear0533** [comment]
> after duplication, should all models be unselected and unhidden?

Yes, that would be preferred. Please feel free to make a PR when your fix is ready. Thank you for catching that!

**Comment by MassKlaus** [comment]
Created the pull request! but this fix is mostly for this thing alone, I think the main issue is more relating to this section here in the code.

![devenv_EhEdoxKCYR](https://github.com/Pear0533/FLVER_Editor/assets/63692278/0f2a75e1-6b36-413a-af4d-319d9ea207e4)

Whatever the value is on the table,  true or false, we still do the same behavior of adding or removing based on if it exists in the indices list or not. I think changing it to only removing on true and only adding on false could fix our odd behavior but I am unsure on how it would interact with the UpdateUI method just yet. Haven't done much reading in the codebase yet

**Comment by MassKlaus** [comment]
What do you think?

**Comment by Pear0533** [comment]
I am already aware that the UpdateIndicesList method (and several other methods in the program code) are not up-to-par, especially when it comes to proper handling for the program state. I re-wrote/adjusted a majority of the code roughly two to three years ago when I wasn't super experienced with programming in general. It would be too much work, for me at least, to re-write a good chunk of the program's code to use clever abstraction along with correct global state handling.

If you have the time and you're willing to, please feel free to re-write areas of the code which need it most. Again, the codebase is definitely not ideal to work with and debug. It would be much better off if code was re-written from scratch where appropriate.

**Comment by Pear0533** [comment]
Many thanks for your contributions! I truly appreciate it.

**Comment by Pear0533** [comment]
What's your name on Discord? I'd like to credit you in the ?ServerName? release notes (I can already credit you in the GitHub release notes since you're marked as a contributor).

**Comment by MassKlaus** [comment]
discord: klaus_ma

There is my discord! 

And regarding the codebase, I think it's been really well made. I didn't have a hard time reading through it nor found much mistakes, it's actually really great work.
